ui/js/graphs.js

```diff
@@ -801,6 +814,71 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
         });
     };
 
+    var resolvePromise = $q.defer();
```

Comment:
resolvePromise probably isn't the best name for this variable, since it just describes what the variable is, without providing any hint on what it's for.

Let's call it loadingExtraDataPromise instead.